Don't touch system-wide files when run as user.

loathingKernel opened the pull request:
When Numix Folders application is run as a regular user, and the themes are installed system-wide, instead of replacing files in system-wide, create a new theme which inherits the base theme specified by the user. This helps with package managers that expect for files installed in `/usr` to remain untouched. Also helps with package upgrades of the base themes.

Although it works, I haven't tested it thoroughly yet, so I consider it work-in-progress. It is quite slow for now, possibly because most string manip is done in bash or inefficiently. Still, I would like to get some more eyes on it.

loathingKernel committed:
Don't touch system wide files when run as user. Instead create a separate theme.

loathingKernel committed:
Fix use without gui

loathingKernel committed:
Keep shellcheck happy

palob commented:
Thanks for your first steps into this direction. It's needed in order to address #101 and #162.

loathingKernel committed:
Simplified and optimized the logic a bit. Added some comments and linted everything.

loathingKernel commented:
I have an issue with large icons (96 and 128 I believe) when using custom colours. If someone more knowledgeable of the theme could take a look at why it is happening, it would be greatly appreciated.

loathingKernel committed:
Fix crash with missing theme value in config

benburrill commented:
Why use `~/.icons` as the destination dir as opposed to the more xdg-ish `~/.local/share/icons`?  Is `~/.icons` more compatible or something?

loathingKernel commented:
There wasn't any specific reason I chose `~/.icons` over the xdg equivalent, just force of habit.

Foggalong reviewed (changes requested):
Very belatedly I'm testing this now and I generally like it other than that it's quite slow during the symlink creation stage. Not sure what part of the changes is the bottleneck there but some testing needed.

Also yeah, `~/.icons` needs switching for `~/.local/share/icons` just to remain standards compliant but that's an easy enough switch.

Foggalong commented:
Bug which arises with this method is that any symlinks not included in the created "Numix Folders" theme aren't being picked up by the inheritance system. Not sure why that's the case but further testing there needed also.

loathingKernel committed:
Use ~/.local/share/icons insteand of ~/.icons

loathingKernel committed:
Replace `basename` with bash expansion

Reduces execution time by ~4 seconds on my system

loathingKernel commented:
The bottleneck is the amount of symlinks and that `find` is invoked all the time. That combined with that fact that it is doing string manipulation in bash (which is still faster than invoking external programs) makes up most of the execution time. I don't think that the current solution as it is can do any better, it will require a change of method, possibly a rewrite. If so, I would suggest to move it to Python or something similar for better code clarity and possibly performance. 

The issue with the custom folder colours not being used in large sized icons is due to those icons being symlinked from the system-wide Numix folder as they don't exist in the Numix-Folders. A solution could be to copy those files, either in the script or by including them in Numix-Folders.

loathingKernel committed:
Don't create folders that exist as symlinks in the base theme.

Instead recreate the linking in the Numix-Folders theme.
Cuts theme creation time by about 50%

loathingKernel committed:
Fix icon sizes larger than 96 being broken for custom colors

Removes 96 directory and symlinks 96 -> 48.
This mirrors base Numix theme where the subfolders of 96 are
symlinks to the 48 equivalents

loathingKernel committed:
Fix a few deprecation warnings the UI

loathingKernel committed:
Revert "Fixed script (#190)"

This reverts commit c213b67a4f846f131d8eb80c00ae04e15dcbe0bb.

Foggalong reviewed (commented):
I am no longer a member of Numix Project so my review is moot (see numixproject/numix-core#5868).
